### ISSUE TITLE
json_parse issue 

### ISSUE THREAD

**Issue by davydav** [eddiezane/sendgrid-actionmailer#70]
Hi there ! I've read the other issues regarding the `json_parse` method, however I think my issue is different : I'm trying to upgrade our project from SendGrid v1 to v3 using your gem; all of our email templates contain fields to be substitued using the handlebar syntax. 
I pass a personalization object looking like this : 
```
personalizations: [{ to: [{ email: 'davy.keliny@plezi.co' }], substitutions: {'{{contact_first_name}}' => contact.first_name } }]
```
But I get this error : 
``` 
JSON::ParserError - 419: unexpected token at '{"to":[{"email":"davy.keliny@plezi.co"}], "substitutions":{"{{contact_first_name}}"":"Davy"}}]':
  json (1.8.6) lib/json/common.rb:155:in 'parse'
/Users/dav/Documents/plezi/vendor/gems/sendgrid-actionmailer-2.6.0/lib/sendgrid_actionmailer.rb:176:in 'json_parse'
  /Users/dav/Documents/plezi/vendor/gems/sendgrid-actionmailer-2.6.0/lib/sendgrid_actionmailer.rb:190:in 'add_personalizations'
  /Users/dav/Documents/plezi/vendor/gems/sendgrid-actionmailer-2.6.0/lib/sendgrid_actionmailer.rb:29:in 'deliver!'
```

I'm trying to understand where the extra double quote character comes from here : 
`{"{{contact_first_name}}"":"Davy"}` but I'm having a little trouble debugging the regexp, the only fix I  managed to come up with is this : 
```
def json_parse(text, symbolize=true)
  JSON.parse(text.empty? ? '{}' : text.gsub(/:*\"*([\%a-zA-Z0-9_-]*)\"*(( *)=>\ *)/) { "\"#{$1}\":" }.gsub(/\"\":/, '":'), symbolize_names: symbolize)
end
```

If you have a cleaner suggestion that would allow me not to use a modified version of  your code, it would be great ! 
Thank you for your great work :)

**Comment by travisp** [comment]
Although I worked on the personalizations feature, this seems similar to https://github.com/eddiezane/sendgrid-actionmailer/issues/66 -- the method of accessing the attributes from the mail object seems to use JSON parsing to preserve backwards compatibility with older mail versions. The `unparsed_value` method wasn't exposed by mail until 2.7.1

To solve this, sendgrid-mailer converts the result from mail's `value` method (which I guess is a string representation of a hash) back into JSON and then parses the JSON to get the hash. It seems a bit error prone, so I wonder if there's a better way to do this (other than requiring >= 2.7.1)

**Comment by tyrauber** [comment]
@travisp This is addressed in 3.0.0. Thanks for raising this issue.
